packages/liveblocks-react-ui/src/components/internal/Attachment.tsx

```diff
@@ -0,0 +1,306 @@
+"use client";
+
+import type {
+  CommentAttachment,
+  CommentLocalAttachment,
+  CommentMixedAttachment,
+} from "@liveblocks/core";
+import { useAttachmentUrl, useIsInsideRoom } from "@liveblocks/react";
+import type {
+  ComponentPropsWithoutRef,
+  KeyboardEvent,
+  MouseEventHandler,
+  PointerEvent,
+} from "react";
+import React, { memo, useCallback, useEffect, useMemo, useState } from "react";
+
+import { CrossIcon } from "../../icons/Cross";
+import { SpinnerIcon } from "../../icons/Spinner";
+import { WarningIcon } from "../../icons/Warning";
+import type { GlobalOverrides } from "../../overrides";
+import { useOverrides } from "../../overrides";
+import { AttachmentTooLargeError } from "../../primitives";
+import { useComposerAttachmentsContextOrNull } from "../../primitives/Composer/contexts";
+import { classNames } from "../../utils/class-names";
+import { formatFileSize } from "../../utils/format-file-size";
+import { Tooltip } from "./Tooltip";
+
+interface FileAttachmentProps extends ComponentPropsWithoutRef<"div"> {
+  attachment: CommentMixedAttachment;
+  onDeleteClick?: MouseEventHandler<HTMLButtonElement>;
+  preventFocusOnDelete?: boolean;
+  overrides?: Partial<GlobalOverrides>;
+}
+
+const IMAGE_PREVIEW_MAX_SIZE = 50 * 1024 * 1024; // 50 MB
+
+const fileExtensionRegex = /^(.+?)(\.[^.]+)?$/;
+
+function splitFileName(name: string) {
+  const match = name.match(fileExtensionRegex);
+
+  return { base: match?.[1] ?? name, extension: match?.[2] };
+}
+
+function getFileAttachmentIconGlyph(mimeType: string) {
+  if (
+    mimeType === "application/zip" ||
+    mimeType === "application/x-rar-compressed" ||
```

Comment:
```suggestion
    mimeType === "application/x-rar-compressed" ||
    mimeType === "application/vnd.rar" ||
    mimeType === "application/x-bzip" ||
    mimeType === "application/x-bzip2" ||
```

Listed here as the rar and bzip mimetypes: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types